### PR TITLE
[Secrets] Improving consistency with other Tier-1 languages

### PR DESCRIPTION
**Pull request by seankane-msft**
APIView link: https://apiview.dev/Assemblies/Review/81f68e700dcd4243a0ced7223e4d71fa

### CODE REVIEW

File: sdk/keyvault/azsecrets/client.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ type SetSecretOptions struct {
 	ContentType *string `json:"contentType,omitempty"`
 
 	// The secret management attributes.
-	SecretAttributes *Attributes `json:"attributes,omitempty"`
+	SecretAttributes *Properties `json:"attributes,omitempty"`
```
Comment (heaths) [inline sdk/keyvault/azsecrets/client.go:119]: Exposed type should be `SecretProperties`. If this type is only used for de/serialziation, shouldn't it be `secretAttributes`?

File: sdk/keyvault/azsecrets/client.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ type SetSecretResponse struct {
 	RawResponse *http.Response
 
 	// The secret management attributes.
-	Attributes *Attributes `json:"attributes,omitempty"`
+	Attributes *Properties `json:"attributes,omitempty"`
```
Comment (heaths) [inline sdk/keyvault/azsecrets/client.go:138]: Same. No type using "attributes" should be public.

File: sdk/keyvault/azsecrets/client.go [hunk 3]
```diff
@@ -397,12 +395,29 @@ func (c *Client) GetDeletedSecret(ctx context.Context, secretName string, option
 type UpdateSecretPropertiesOptions struct {
 	// Version is the specific version of a Secret to update. If not specified it will update the most recent version.
 	Version string
+
+	// Type of the secret value such as a password.
+	ContentType *string `json:"contentType,omitempty"`
+
+	// The secret management attributes.
+	SecretAttributes *Properties `json:"attributes,omitempty"`
```
Comment (heaths) [inline sdk/keyvault/azsecrets/client.go:403]: `SecretProperties` (won't repeat, so please check all occurrences).

**Review by heaths — CHANGES_REQUESTED** [review]

File: sdk/keyvault/azsecrets/client.go (hunk 3 above)
Comment (seankane-msft) [reply sdk/keyvault/azsecrets/client.go:403]: Back to the stuttering discussion from other PRs, I'm going to keep this as `Properties`.

**Review by heaths — APPROVED** [review]

**Pull request merged**